src/DepositHandler.sol

```diff
@@ -42,6 +47,31 @@ contract DepositHandler is Pausable, AccessControl, IDepositHandlerErrors {
         uint256 withdrawAmount
     );
 
+    /**
+     * @dev Checks if user has a required status to do a deposit/withdraw. 
+     */ 
+    modifier isAllowed(Status _status) {
+        Status status = deposits[msg.sender].status;
+        if (status != _status) {
+            revert DepositHandler__NotAllowedActionWithYourStatus();
+        }
+        _;
+    }
+
+    /**
+     * @dev Checks if user has a required status to do a deposit/withdraw.
+     * It is '<' instead of '<=', because I assume that encode team/manager need some time after
+     * the final day of the bootcamp to sum up all participants results. Like a day after a bootcamp users
+     * will be able to withdraw their deposits. 
+     */
+    
+    modifier whenFinished {
```

Comment:
if there is an unwanted scenario where encode creates a bootcamp with a duration too long and by mistake shares this info to the participants, they might deposit and have their funds locked, until this modifier allows withdraw. we should allow some emergencyWithdraw to mitigate this edge case or consider whether handling through the Status enum is enough (no reliance on the timestamps, just Encode to clear the participant to be eligible for withdrawal)

Reply:
Agree, having an emergency withdraw is possible and necessary. I'll cover that scenario in contract. Thx